### ISSUE TITLE
[bug] TabViewAnimated initialLayout modifies layout after first render

### ISSUE THREAD

**Issue by ixje** [react-native-community/react-native-tab-view#297]
The title might not be 100% accurate, but that's because I don't fully understand what the initialLayout is used for in the whole code base. Regardless, there is some weird stuff happening to my layout that's flex based.

For clarity's sake I've given each element in the Row component some background colours such that it's easy to see what's happening to the layout.

Note: I've only tested this on iOS as I currently don't have a working Android env.
 
### Expected behaviour
The yellow/blue/grey box sizes do not change when adding an item
![expected](https://thumbs.gfycat.com/SereneConcernedFrilledlizard-size_restricted.gif)
### Actual behaviour
The yellow/blue/grey box sizes do change when adding an item
![actual](https://thumbs.gfycat.com/ExcitableSerpentineBluebottle-size_restricted.gif)

### Code sample, screenshots
See above

### What have you tried
The only difference between expected and actual behaviour is specifying an `initialLayout` with a width of 1 to `TabViewAnimated` as such 

```
            <TabViewAnimated
                style={styles.container}
                navigationState={this.state}
                renderScene={this._renderScene}
                renderHeader={this._renderHeader}
                onRequestChangeTab={this._handleChangeTab}
                initialLayout={{ height: 0, width: 1 }} // must set width to 1 to fix some weird bug in flex layouting.
            />
```

This means I've got my work around, but it might be worth investigating why an initialLayout with width:0 (as set by default at [TabViewAnimated.js#L84](https://github.com/react-native-community/react-native-tab-view/blob/8ed6baaf9e1fcb118780bf1538ffafe88879ef1a/src/TabViewAnimated.js#L84)) causes this behaviour.

**Comment by leethree** [comment]
I'm seeing the same bug when using TabNavigator and TabViewPagerPan in `react-navigation`. It seems the default `width=0` is problematic.

**Comment by ikzjfr0** [comment]
If you don't provide an exact width,  you could see tab bar indicator jump back and forth, similar like this one: https://github.com/react-native-community/react-native-tab-view/issues/493

The root cause is: 
if you provide width, then `this.props.layout.width` = 0, then `initialOffset` and `_normalizeScrollValue` calculation is not right,  which both are used to determine tab bar and indicator offset value.

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
